### DISCUSSION TITLE
Re-use value of _answers-file

### DISCUSSION THREAD

**Discussion by nilsdebruin** [copier-org/copier#714] [Q&A] [answered]
Hi all!

I am experimenting with copier and I am really loving it! I have the following question:

When you create a template and use the default answer file filename, I see the following happening when I run it:

`copier <template_dire> <target_dir>` --> all questions are asked and default values are filled in

When I rerun the same command,  all questions that have been previously answered are filled in automatically (this is being parsed from the answers file).

When I now change the copier.yml file and include the following _answers-file with a different filename, this results in

`copier <template_dire> <target_dir>` --> all questions are asked and default values are filled in

When I rerun the same command, it asks for all questions are asked and default values are filled in.

When I supply an extra command line switch copier can read the answers:

`copier <template_dir> <target_dir> -a <answers_file_name>`

It doesn't seem to pick up the answer file setting from the copier.yml when re-running the command.

It would be great if copier would read the answers from the filename of the changed _answers-file setting. Is this possible or am I doing something wrong / missing something?

If it is unclear, I will add a more elaborate example!

**Comment by yajo** [comment, marked as answer]
I'm not sure if I understand you correctly, but if you're wanting to use a different answers file name, then:

1. When you copy the template for the 1st time, it will take the default defined in `copier.yml` if `-a` is not used in the CLI.
2. For next updates, you need to specify the answers file you're updating from (because it's not the default one): `copier -a .custom-answers.yml update`

This is expected behaviour, because I can tell Copier what's the default value for an answers file, but once it's recorded, Copier cannot know which template are you updating unless you point it to an answers file (where that info is recorded).

**Reply by nilsdebruin** [reply]
Hi @yajo thanks for your quick answer!

As I am now developing a template, I would suspect that copier would read the copier.yml file, extract the _answers-file value and use this as the answer file name when repeating the creation process. This seems to be happening when the default answer file name is used (all answers are found automatically in the default answers file).

I have created a repo containing example code which can be found here:

https://github.com/nilsdebruin/copier-example

1. If you run `copier_create_example.sh` as is, the template is created in the subfolder copier-tests with the name example (I have excluded this in the copier.yml so you don't get a recursion error).
2. A question about the project name is asked. Just fill in something random
3. You get the output you would expect.
4. Re-run `copier_create_example.sh`, when the project name question is asked, the answer from the previous run is shown
5. Now delete the copier-tests/example folder
6. Uncomment line 10 in copier.yml: `# _answers-file: .copier-answers-example.yml`
7. Run `copier_create_example.sh`, again fill in something random as name
8. Re-run `copier_create_example.sh`, the project name is not filled in automatically, although this had been done in step 7
9. Now uncomment the last part of line 1 in copier_create_example.sh: `copier . copier-tests/example -a .copier-answers-example.yml`
10. If you now re-run `copier_create_example.sh`, the answer is filled in again

I now understand that this is expected behaviour, but when I encountered it, I couldn't understand it! I have put up above repo and explanation for possible future reference. Thanks again for your answer!
